src/v0/destinations/salesforce/utils.js

```diff
@@ -27,7 +27,11 @@ const salesforceResponseHandler = (destResponse, sourceMessage, authKey) => {
   const { status, response } = destResponse;
 
   // if the response from destination is not a success case build an explicit error
-  if (!isHttpStatusSuccess(status) && status >= 400) {
+  if (
+    // This condition will pop-up when `invalid_grant` error is thrown from Salesforce
+    !Number.isInteger(status) ||
```

Comment:
should not we try to convert to number and see and if can't covert then we can consider as error.

Reply:
This would throw the error when `status` for some reason is not a number.